testing/GenericCollectionTest.php

```diff
@@ -563,4 +563,25 @@ public function testGetClass(): void
         $testCollection = new GenericCollection();
         $this->assertEquals('JTL\Generic\GenericCollection', $testCollection->getClass());
     }
+
+    public function testCanChunk(): void
+    {
+        $pieces = [];
+        $sut = new GenericCollection('string');
+        for ($i = 0; $i < 10; $i++) {
+            $pieces[$i] = uniqid();
+            $sut->add($pieces[$i]);
+        }
+
+        self::assertSame(10, $sut->count());
+        $chunkList = $sut->chunk(2);
+        self::assertCount(5, $chunkList);
+        foreach ($chunkList as $key => $chunk) {
+            self::assertInstanceOf(GenericCollection::class, $chunk);
+            self::assertSame(2, $chunk->count());
+            foreach ($chunk as $chunkKey => $piece) {
+                self::assertSame($pieces[($key*2)+$chunkKey], $piece);
```

Comment:
Hard to read and understand - and in case of a error hard to debug.

Suggestion
assert the elements without using foreach stuff. Do it manual the next Dev will thank you ;)
Maybe split it by 4 so you get the last chunk smaller as expected